core/controllers/base_test.py

```diff
@@ -593,3 +595,64 @@ def test_get_items(self):
                 'param2=value%20with%20%26%20%2B%20-%20/&'
                 'param3=value%20with%20.%20%%20@%20123%20=%20!%20%3C%3E')
             self.assertDictContainsSubset(params, result)
+
+
+class ControllerClassNameTest(test_utils.GenericTestBase):
+
+    def test_controller_class_names(self):
+        """This function checks that all controller class names end with
+        either 'Handler' or 'Page'.
+        """
+        for url in main.URLS:
+            if not isinstance(url, tuple):
+                handler = str(url.handler)[8:-2].rsplit('.', 1)[1]
+                class_path = str(url.handler)[8:-2].rsplit('.', 1)[0]
+                file_name = class_path.replace('.', '/') + ".py"
+                module = importlib.import_module(class_path)
+                clazz = getattr(module, handler)
+                all_base_classes = [base_class.__name__ for base_class in
+                                    (inspect.getmro(clazz))]
+                if 'BaseHandler' in all_base_classes:
+                    class_return_type = clazz.GET_HANDLER_ERROR_RETURN_TYPE
```

Comment:
Ensure there's a check as well for "any class with a get handler has a get_handler_error_return_type that's one of the allowed values"